### ISSUE TITLE
Zlib crashes app

### ISSUE THREAD

**Issue by JoshZA** [mzabriskie/axios#287]
Heya

I'm getting errors like this in production - my code is loading pages from wild http servers, so this is most likely related to them being buggy.

```
{ [Error: unexpected end of file] errno: -5, code: 'Z_BUF_ERROR' }
Error: unexpected end of file
    at Zlib._handle.onerror (zlib.js:363:17)
```

This is on node 4x and v5.7.0

Would it be possible to add a ```stream.on('error'``` clause in axios\lib\adapters\http.js and throw this as an exception to the axios response promise?

**Comment by nickuraltsev** [comment]
Yes, this needs to be handled. I'd be happy to accept a PR.

**Comment by nickuraltsev** [comment]
Fixed by #303.

**Comment by selfagency** [comment]
I started to suddenly have this same issue beginning at 5 o'clock this morning in my production environment. No changes were pushed from my codebase, no package upgrades were made, but all of a sudden every request sent through Axios returns:

```
{"name":"nylon","hostname":"ip-XXX-XXX-XXX","pid":23352,"level":50,"err":{"message":"incorrect header check","name":"Error","stack":"Error: incorrect header check\n    at Unzip.zlibOnError (zlib.js:153:15)","code":"Z_DATA_ERROR"},"msg":"incorrect header check","time":"2018-06-12T19:27:28.304Z","v":0}
```

The only requests are being sent to my API which also did not change.
